### PR TITLE
Introduction of a system to feed displays with the correct charsets. Example in language_de.h

### PR DESCRIPTION
**Pull request by AnHardt**
Please have a look on the last three commits.

I still wand to work on the actual German wording with you.
Please comment on the separate  commits.

### CODE REVIEW

**Comment by AnHardt** [comment]
< copied for reference from #1430  >

We can have some more or less good Umlaute on the U8glib displays out of the box

#if DISPLAY_CHARSET==3 // U8glib
#define STR_Ae "\304" // yes
#define STR_ae "\344" // yes
#define STR_Oe "\326" // yes
#define STR_oe "\326" // no
#define STR_Ue "\334" // yes
#define STR_ue "\334" // no
#define STR_sz "\337" // yes
#define STR_Deg "\260" // yes
#endif

Sadly mussing 'ö' and ü''

\000 - \077


\100 - \177


\200 - \277


\300 - \377

**Comment by AnHardt** [comment]
For reference.
My SainSmart REPRAP_DISCOUNT_SMART_CONTROLLER

jap000-077
jap000-177

jap100-177
jap100-177

jap200-277
jap200-277

jap300-377
jap300-377

**Comment by AnHardt** [comment]
Any wishes what to add besides "ÄäÖöÜüß°" and thermometer?

**Comment by thinkyhead** [comment]
Good stuff! I am looking it over now. +1 on your `MSG_ENDSTOP_ABORT` comment.

File: Marlin/language.h [hunk 1]
```diff
@@ -160,6 +162,71 @@
...
+#define STR_sz "ss"
+#define STR_Deg ""
+#define STR_THERMOMETER "\302"
+
+#ifdef DISPLAY_CHARSET_DOGM
+  #define STR_Ae "\304" // U8glib
+  #define STR_ae "\344"
+  #define STR_Oe "\326"
+  #define STR_oe STR_Oe
+  #define STR_Ue "\334"
```
Comment (thinkyhead) [inline Marlin/language.h:182]: We can drop these defines that do nothing, I suppose!

**Comment by thinkyhead** [comment]
I like it! Sensible defaults, allow the user to choose something else. And documents all these displays in a concise way. A great contribution, thank you.

**Pull request merged**